client/src/http_sender.rs

```diff
@@ -122,3 +148,22 @@ impl RpcSender for HttpSender {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn http_sender_on_tokio_multi_thread() {
+        let http_sender = HttpSender::new("http://localhost:1234".to_string());
+        let _ = http_sender.send(RpcRequest::GetVersion, serde_json::Value::Null);
+    }
+
+    #[tokio::test(flavor = "current_thread")]
+    #[should_panic(expected = "can call blocking only when running on the multi-threaded runtime")]
+    async fn http_sender_ontokio_current_thread_should_panic() {
```

Comment:
This isn't great, but I don't see a way around.  Fortunately `[tokio::main]` creates a `multi_thread` runtime, so most users won't see this (but `#[tokio::test]`, sans flavour, is `current_thread`)

Reply:
I see, obv not ideal but okay for now. I haven't thought of anything better.